static_frame/core/index.py

```diff
@@ -757,9 +757,12 @@ def _index_iloc_map(self: I, other: I) -> TNDArrayAny:
 
         Equivalent to: self.iter_label().apply(other._loc_to_iloc)
         '''
-        if self.__len__() == 0:
+        if self.__len__() == 0 or other.__len__() == 0:
             return EMPTY_ARRAY
 
+        if self.dtype == DTYPE_OBJECT or other.dtype == DTYPE_OBJECT:
+            return self.iter_label().apply(other._loc_to_iloc, dtype=DTYPE_INT_DEFAULT)  #type: ignore [no-any-return]
+
```

Comment:
@flexatone Sorry for the comment spam. This was my solution to the NaN issue. If either Index being compared are object dtypes, I fall back on the default behavior. I only use the optimized numpy route if the dtypes are defined